### PR TITLE
Move to uv as package manager

### PR DESCRIPTION
**Pull request by LGonzalezGomez**
Since we are in 2024 we might as well go to uv directly. I created a toml but we should check config.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,20 +15,19 @@ coverage_xml:
...
-	python -m pip install -e ".[dev]"
+	python -m pip install uv
+	uv sync --extra "dev"
 	pre-commit install
 
 install-test:
 	pip install --upgrade pip setuptools wheel
-	python -m pip install -e ".[test]"
-
-install-only-test:
```
Comment (LGonzalezGomez) [inline Makefile:29]: @david26694 Is this needed? Seems weird to me and is a bit weird doing it in toml file

**Comment by LGonzalezGomez** [comment]
Version can be found and downloaded here
https://test.pypi.org/project/cluster-experiments/0.20.1/

Must still fix GHA to use uv

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -16,5 +16,6 @@ jobs:
       - run: cp README.md docs/index.md
       - run: cp -r theme docs/theme
       - run: make install-dev
+      - run: source .venv/bin/activate
```
Comment (david26694) [inline .github/workflows/docs.yml:19]: if we don't do this, does it not work?

File: setup.py [hunk 3]
```diff
@@ -1,71 +1,5 @@
 from setuptools import find_packages, setup
```
Comment (david26694) [inline setup.py:1]: do we need a setup.py now?

**Review by david26694 — COMMENTED** [review]
Couple comments, but apart from that, let's just change the readme to require uv for contributing and merge

File: Makefile (hunk 1 above)
Comment (david26694) [reply Makefile:29]: not really

**Review by david26694 — APPROVED** [review]

**Pull request merged**